e2e-app/src/app/modal/stack/modal-stack.component.ts

```diff
@@ -0,0 +1,33 @@
+import {Component, TemplateRef} from '@angular/core';
+import {NgbActiveModal, NgbModal} from '@ng-bootstrap/ng-bootstrap';
+import {Observable} from 'rxjs';
+import {map} from 'rxjs/operators';
+
+@Component({templateUrl: './modal-stack.component.html'})
+export class ModalStackComponent {
+  constructor(private modalService: NgbModal) {}
+
+  openModal(content: TemplateRef<any>) { this.modalService.open(content); }
+
+  search = (text$: Observable<string>) => text$.pipe(map(() => ['one', 'two', 'three']));
+
+  openSecondModal() { this.modalService.open(NgbdStackedModal); }
```

Comment:
can't you simply reuse `openModal()` and remove `NgbdStackedModal` component (use templates for both first and second modals) ? It would be much simpler.